### PR TITLE
Indented most of the source code

### PR DESCRIPTION
**Pull request by Richard-Degenne**
This PR is a follow-up of PR #400 about issue #393.

Most of the `ml` files of the source folder have undergone an `ocp-indent` command, followed by a quick review. Some remarks about this:

- Globally, the indentation was consistant within files, but varied a bit between files;
- Most of the changes are on pattern matches;
- I've left `core/lwt.ml` and `unix/lwt_unix.cppo.ml` untouched, so as not to cause conflicts with ongoing PRs;
- `ocp-indent`ing `mli` files wasn't really necessary, and it kept breaking comments indentation;
- `ocp-indent` got the camlp4 and ppx files pretty wrong, and I indented them mostly by hand. Since they are quite obscure to me, these ones may require some extra attention.

### CODE REVIEW

**Comment by raphael-proust** [comment]
AFAIAC, there are no PRs locking `unix/lwt_unix.cppo.ml` right now.

There are probably going to be some changes in the short term (e.g., #408) but there are no PR for these.

**Comment by aantron** [comment]
Thanks! I'll review this in a bit. Just want to say that I'm minutes or hours away from submitting a PR for #408, but it won't touch the `.ml` file, at least for now. So, I think you can add `lwt_unix.cppo.ml` as @raphael-proust suggests (thanks again) :)

File: src/preemptive/lwt_preemptive.ml [hunk 1]
```diff
@@ -207,20 +207,20 @@ let detach f args =
...
-        (* Put back the worker to the pool: *)
-        add_worker worker
-      else begin
-        decr threads_count;
-        (* Or wait for the thread to terminates, to free its associated
-           resources: *)
-        Thread.join worker.thread
-      end;
-      Lwt.return_unit)
+       if worker.reuse then
```
Comment (aantron) [inline src/preemptive/lwt_preemptive.ml]: IMO ocp-indent gets this kind of thing wrong. It shouldn't cause an extra space if you have to add a `(` to a function (or any other expression). Is it easy to fix in the ocp-indent settings? If not, it's fine – no need to go through all these files. The PR does much more good than harm, even with this nit.

File: src/unix/lwt_log.ml [hunk 2]
```diff
@@ -244,47 +244,47 @@ let syslog ?(template="$(date) $(name)[$(pid)]: $(section): $(message)") ?(paths
...
-        syslog_socket := Some x;
-        Lwt.return x
+      syslog_connect paths >>= fun x ->
+      syslog_socket := Some x;
+      Lwt.return x
   in
   make
     ~output:(fun section level lines ->
-               Lwt_mutex.with_lock mutex
+      Lwt_mutex.with_lock mutex
```
Comment (aantron) [inline src/unix/lwt_log.ml]: I'm really glad things like this are getting fixed.

File: src/preemptive/lwt_preemptive.ml (hunk 1 above)
Comment (Richard-Degenne) [inline src/preemptive/lwt_preemptive.ml]: Hmm, I guess it does this to align everything two spaces after the `fun` keyword, which is one column after the beginning of the line. I agree it looks kinda wrong. I'll see what I can do.

File: src/util/configure.ml [hunk 3]
```diff
@@ -22,26 +22,26 @@ let args = [
...
-   *                     the camlp4 directory.
-   * '-use-camlp4 true' will write an empty file which does nothing
-   *
-   * This is a workaround required to overcome some weird camlp4 packaging behaviour
-   * where ocamlfind sometimes installs a dummy META file for it even though it's
-   * not actually installed. *)
-  let f = open_out "src/jbuild-ignore" in
-  (if !use_camlp4 = Some(true) then () else Printf.fprintf f "camlp4");
-  close_out f
+    let f = open_out "src/unix/lwt_config" in
```
Comment (aantron) [inline src/util/configure.ml]: Looks like ocp-indent got this wrong. I wonder why? I don't think this should be indented more than the previous line (am I missing something?).

**Comment by aantron** [comment]
So most of it looks good – thanks for doing this. The only other nit I have is that the glib commit should be titled something like ``Fixed `glib` indentation`` like all the others are, so people can click past it in blame.

File: src/preemptive/lwt_preemptive.ml (hunk 1 above)
Comment (Richard-Degenne) [inline src/preemptive/lwt_preemptive.ml]: Available configuration doesn't seem to provide an option for this behavior.

Is it worth opening an issue on their side?

Comment (aantron) [inline src/preemptive/lwt_preemptive.ml]: Sure, they might be able to answer it, fix it, or teach me/us why the ocp-indent way is the right way :) Also cc @vasilisp for any experience with this.

I don't think this blocks the PR though, unless you want to wait for it.

File: src/util/configure.ml (hunk 3 above)
Comment (Richard-Degenne) [inline src/util/configure.ml]: There's a similar issue reported [here](https://github.com/OCamlPro/ocp-indent/issues/229). I'll keep an eye on it.

Comment (aantron) [inline src/util/configure.ml]: Thanks, subscribed to that issue.

**Comment by aantron** [comment]
Ok, looks great. I'm happy to have this in. I'll merge once the builds finish, but they will, of course, succeed.

**Pull request merged**